### PR TITLE
KVS API template for storing vector metadata

### PR DESCRIPTION
**Pull request by smorihira**
- Implemented a template for the KVS API in Rust to store vector metadata

### Description
- Implemented `meta.proto` (`get`, `set`, `delete`) and part of `payload.proto`
- Created a Rust template for the implementation of the traits generated by the proto (`rust/bin/meta`)

### Versions

- Vald Version: v1.7.13
- Go Version: v1.23.0
- Rust Version: v1.80.0
- Docker Version: v27.1.1
- Kubernetes Version: v1.30.3
- Helm Version: v3.15.3
- NGT Version: v2.2.4
- Faiss Version: v1.8.0

### Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/main/CONTRIBUTING.md) document and completed [our CLA agreement](https://cla-assistant.io/vdaas/vald).
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?

## Summary by CodeRabbit

- **New Features**
	- Introduced a `Meta` component with methods for managing metadata (Get, Set, Delete).
	- Added comprehensive API documentation for the `Meta` component, enhancing clarity for developers.
	- Implemented a gRPC service for the `Meta` component, facilitating efficient metadata management.

- **Bug Fixes**
	- Improved error handling in the metadata service, ensuring more informative responses.

- **Documentation**
	- Expanded API documentation to include detailed descriptions of new metadata structures and operations.

- **Chores**
	- Updated project dependencies to enhance compatibility and performance.

### CODE REVIEW

File: rust/bin/meta/src/handler.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+mod meta;
+
+#[derive(Default, Debug)]
+pub struct Meta {
+    
+}
+
+
+
+
```
Comment (hlts2) [inline rust/bin/meta/src/handler.rs:25]:
```suggestion
```

File: rust/Cargo.toml [hunk 2]
```diff
@@ -17,8 +17,9 @@
...
 	"libs/proto",
-	"bin/agent",
-	"libs/algorithm",
-	"libs/algorithms/ngt",
-	"libs/algorithms/faiss",
+	"bin/meta",
+	# "bin/agent",
+	# "libs/algorithm",
+	# "libs/algorithms/ngt",
+	# "libs/algorithms/faiss", "bin/meta",
```
Comment (hlts2) [inline rust/Cargo.toml:24]:
```suggestion
	"bin/agent",
	"libs/algorithm",
	"libs/algorithms/ngt",
	"libs/algorithms/faiss",
```

**Review by hlts2 — APPROVED** [review]
LGTM!

**Review by kpango — APPROVED** [review]
LGTM

**Comment by kpango** [comment]
https://github.com/all-contributors please add @smorihira for design

**Pull request closed without merging**